colibris/views/base.py

```diff
@@ -37,4 +37,5 @@ def __init__(cls, name, bases, attrs):
 
 
 class View(web.View, metaclass=ViewMeta):
-    pass
+    required_authentication = None
+    required_permissions = None
```

Comment:
For `required_permissions`, for me sounds better just `permissions`. And as default, shouldn't it be an empty iterable?

For `required_authentication` I would prefer `needs_authentication`, `authentication_required` or `authentication_is_required`.  `required_authentication` doesn't seem to expect a boolean value.

Reply:
> For `required_permissions`, for me sounds better just `permissions`.
> For `required_authentication` I would prefer `needs_authentication`, `authentication_required` or `authentication_is_required`.  `required_authentication` doesn't seem to expect a boolean value.

Alright, let's go with `permissions` and `authentication_required`.

> And as default, shouldn't it be an empty iterable?

In the context of this PR, empty iterable permissions don't make too much sense anymore. Either you want to specify permissions or you don't. So no, I wouldn't use empty iterable at all for permissions anywhere across the project.

Reply:
Also, for `required_authentication`, does it make to have a third state (`None`)? Shouldn't it be `True` or `False` by default?

Reply:
As we agreed on this one, `None` would mean "not set". This allows Colibris to automatically switch it to `True` if permissions are specified and to `False` otherwise.

In turn, this allows for cleaner "public" views by default and regular permission-based views without explicitly altering the `authentication_required` flag.